infra/token-transfer-server/src/lib/balance.js

```diff
@@ -44,39 +39,80 @@ async function hasBalance(userId, amount, currentTransferId = null) {
     `User ${user.email} unlocked earnings from lockups`,
     lockupEarnings.toString()
   )
-  // Sum amount withdrawn or pending in transfers excluding the current transfer
-  // if provided (balance check while exeucting transfer)
-  const transferWithdrawnAmount = calculateWithdrawn(
-    user.Transfers.filter(t => t.id !== currentTransferId)
-  )
+  // Sum amount withdrawn or pending in transfers
+  const transferWithdrawnAmount = calculateWithdrawn(user.Transfers)
   logger.debug(
     `User ${user.email} pending or transferred tokens`,
     transferWithdrawnAmount.toString()
   )
+
   // Sum locked by lockups
   const lockedAmount = calculateLocked(user.Lockups)
   logger.debug(`User ${user.email} tokens in lockup`, lockedAmount.toString())
 
+  const nextVestLocked = calculateNextVestLocked(user.Lockups)
+  logger.debug(
+    `User ${user.email} tokens in early lockup`,
+    nextVestLocked.toString()
+  )
+
   // Calculate total available tokens
   const available = vested
     .plus(lockupEarnings)
     .minus(transferWithdrawnAmount)
     .minus(lockedAmount)
+    .plus(nextVestLocked)
```

Comment:
should this be .plus(earlyLockupEarnings) ?

Reply:
When an early lockup is matured (i.e. current date is passed the date stored in "end" column) it doesn't get treated differently to a non-early lockup, so the lockupEarnings includes earnings from early lockups

Ref https://github.com/OriginProtocol/origin/blob/55c8f5803e3d9708a7a826a558cab712c3b5b008/infra/token-transfer-server/src/shared.js#L62-L73

Reply:
Ah thanks. that makes sense re calculateUnlockedEarnings.

Now sorry to be super slow, but why do we need to add nextVestLocked to the available balance? aren't the nextVestLocked tokens locked and by definition not available?

Reply:
That is because `lockedAmount` is subtracted, and it includes early lockups as well. Early lockups are against a future vest and don't change the current balance, so that number needs to be added back in.

I'm not very happy with the naming, I'll see if I can perhaps separate `lockedAmount` into early and standard to make it clearer.

Reply:
Ah I get it now. Thanks for explaining... Maybe add a comment in the code for dummies like me.

Reply:
No it is way too confusing, I've changed it to be clearer. It doesn't include it in the lockedAmount anymore and I'll handle it separately.